src/mcts/search.cc

```diff
@@ -187,45 +187,77 @@ void Search::SendMovesStats() const {
   const float parent_q =
       -root_node_->GetQ(0) -
       kFpuReduction * std::sqrt(root_node_->GetVisitedPolicy());
+  const float U_coeff =
+      kCpuct * std::sqrt(std::max(root_node_->GetChildrenVisits(), 1u));
+
   for (Node* child : root_node_->Children()) {
     nodes.emplace_back(child);
   }
   std::sort(nodes.begin(), nodes.end(),
-            [](const Node* a, const Node* b) { return a->GetN() < b->GetN(); });
+           [&parent_q, &U_coeff](const Node* a, const Node* b) {
+             return
+                 std::forward_as_tuple(a->GetN(), a->GetQ(parent_q) + U_coeff * a->GetU())
+               < std::forward_as_tuple(b->GetN(), b->GetQ(parent_q) + U_coeff * b->GetU());
+           });
 
   const bool is_black_to_move = played_history_.IsBlackToMove();
   ThinkingInfo info;
   for (const Node* node : nodes) {
     std::ostringstream oss;
     oss << std::fixed;
+
     oss << std::left << std::setw(5)
         << node->GetMove(is_black_to_move).as_string();
+
     oss << " (" << std::setw(4) << node->GetMove().as_nn_index() << ")";
-    oss << " N: ";
-    oss << std::right << std::setw(7) << node->GetN() << " (+" << std::setw(2)
-        << node->GetNInFlight() << ") ";
-    oss << "(V: " << std::setw(6) << std::setprecision(2) << node->GetV() * 100
-        << "%) ";
+
+    oss << " N: " << std::right << std::setw(7) << node->GetN() << " (+"
+        << std::setw(2) << node->GetNInFlight() << ") ";
+
+    oss << "(V: ";
+    // Handrolled optional only has a few overloads available, simple usage only
+    optional<float> v = GetCachedFirstPlyValue(node);
+    if (!v && node->IsTerminal()) v = node->GetTerminalNodeValue();
+    if (v) {
+      oss << std::setw(6) << std::setprecision(2) << *v * 100;
+    } else {
+      oss << " --.--";
+    }
+    oss << "%) ";
+
     oss << "(P: " << std::setw(5) << std::setprecision(2) << node->GetP() * 100
         << "%) ";
+
     oss << "(Q: " << std::setw(8) << std::setprecision(5)
         << node->GetQ(parent_q) << ") ";
+
     oss << "(U: " << std::setw(6) << std::setprecision(5)
-        << node->GetU() * kCpuct *
-               std::sqrt(std::max(node->GetParent()->GetChildrenVisits(), 1u))
-        << ") ";
+        << U_coeff * node->GetU() << ") ";
 
     oss << "(Q+U: " << std::setw(8) << std::setprecision(5)
-        << node->GetQ(parent_q) +
-               node->GetU() * kCpuct *
-                   std::sqrt(
-                       std::max(node->GetParent()->GetChildrenVisits(), 1u))
-        << ") ";
+        << node->GetQ(parent_q) + U_coeff * node->GetU() << ") ";
+
     info.comment = oss.str();
     info_callback_(info);
   }
 }
 
+optional<float> Search::GetCachedFirstPlyValue(const Node* node) const {
+  assert(node->GetParent() == root_node_);
+  // This function serves a specific purpose, fetching a NN value from a first
```

Comment:
This comment will probably not be discovered, as it's not expected that function with this name will have a comment about something else.
I'd either delete that comment completely or stated in one line.